confuse.py

```diff
@@ -134,36 +135,193 @@ def __init__(self, filename, reason=None):
 
 # Views and sources.
 
+
+UNSET = object()  # sentinel
+
+
+def _load_first(func):
+    '''Call self.load() before the function is called - used for lazy source
+    loading'''
+    def inner(self, *a, **kw):
+        self.load()
+        return func(self, *a, **kw)
+
+    try:
+        return wraps(func)(inner)
+    except AttributeError:
+        # in v2 they don't ignore missing attributes
```

Comment:
For a little more context: what does the `AttributeError` on Python 2 look like?

Reply:
```
Traceback (most recent call last):
  File "/home/travis/build/beetbox/confuse/.tox/py27-test/lib/python2.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/home/travis/build/beetbox/confuse/.tox/py27-test/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/home/travis/build/beetbox/confuse/.tox/py27-test/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/travis/build/beetbox/confuse/example/__init__.py", line 3, in <module>
    import confuse
  File "/home/travis/build/beetbox/confuse/confuse.py", line 157, in <module>
    class ConfigSource(dict):
  File "/home/travis/build/beetbox/confuse/confuse.py", line 213, in ConfigSource
    __getitem__ = _load_first(dict.__getitem__)
  File "/home/travis/build/beetbox/confuse/confuse.py", line 145, in _load_first
    @wraps(func)
  File "/opt/python/2.7.15/lib/python2.7/functools.py", line 33, in update_wrapper
    setattr(wrapper, attr, getattr(wrapped, attr))
AttributeError: 'method_descriptor' object has no attribute '__module__'
```
In py3, they skip missing attributes.